classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -87,6 +119,107 @@ public function save_action( ActionScheduler_Action $action, \DateTime $date = n
 		}
 	}
 
+	/**
+	 * Helper function to build insert query.
+	 *
+	 * @param array $data Row data for action.
+	 * @param bool  $unique Whether the action should be unique.
+	 *
+	 * @return string Insert query.
+	 */
+	private function build_insert_sql( array $data, $unique ) {
+		global $wpdb;
+		$columns      = array_keys( $data );
+		$values       = array_values( $data );
+		$placeholders = array_map( array( $this, 'get_placeholder_for_column' ), $columns );
+
+		$table_name = ! empty( $wpdb->actionscheduler_actions ) ? $wpdb->actionscheduler_actions : $wpdb->prefix . 'actionscheduler_actions';
+
+		$column_sql                  = '`' . implode( '`, `', $columns ) . '`';
+		$placeholder_sql             = implode( ', ', $placeholders );
+		$where_clause = $this->build_where_clause_for_insert( $data, $table_name, $unique );
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- $column_sql and $where_clause are already prepared. $placeholder_sql is hardcoded.
+		$insert_query                = $wpdb->prepare(
+			"
+INSERT INTO $table_name ( $column_sql )
+SELECT $placeholder_sql FROM DUAL
```

Comment:
So `DUAL` provides backward compatibility with MySQL 5.5. After that, it wasn't needed to specify `DUAL` (MySQL will assume it), but since this is a pretty low effort to get some back-compat I added it.